### PR TITLE
Add 349 release notes

### PR DESCRIPTION
**Pull request by martint**
Closes https://github.com/trinodb/trino/issues/6339

### CODE REVIEW

File: presto-docs/src/main/sphinx/release/release-349.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Release 349 (XX Dec 2020)
+
+## General Changes
+
+* Improve query performance by reducing worker to worker communication overhead. ({issue}`6283`)
+* Improve performance of queries that contain `IS NOT DISTINCT FROM` join predicates. ({issue}`6404`)
+* Fix failure when restricted columns have column masks. ({issue}`6017`)
+* Fix http client distribution JMX metrics. ({issue}`6453`)
```
Comment (dain) [inline presto-docs/src/main/sphinx/release/release-349.md:8]: Should this be "Add"?  Did we remove this by accident and add back, or just expose them correctly?

Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-349.md:8]: These seem to have been broken for ~3 years, so let's change to
```markdown
* Add HTTP client JMX metrics. ({issue}`6453`)
```

File: presto-docs/src/main/sphinx/release/release-349.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# Release 349 (XX Dec 2020)
+
+## General Changes
+
+* Improve query performance by reducing worker to worker communication overhead. ({issue}`6283`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-349.md:5]: Add `#6349` which also improves exchange performance

File: presto-docs/src/main/sphinx/release/release-349.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+* Fix incorrect results for `count(*)` queries with views. ({issue}`5635`)
+
+## Cassandra Connector Changes
+
+* Support `DELETE` statement with primary key or partition key. ({issue}`4059`)
+
+## Elasticsearch Connector Changes
+
+* Improve query analysis performance when Elasticsearch contains many index mappings ({issue}`6368`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-349.md:31]: Missing period

File: presto-docs/src/main/sphinx/release/release-349.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+## Kafka Connector Changes
+
+* Support Kafka Schema Registry for AVRO topics. ({issue}`6137`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-349.md:35]: Capitalize as Avro

**Review by dain — APPROVED** [review]

**Pull request merged**